api4/user.go

```diff
@@ -915,7 +915,15 @@ func updateUserAuth(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(user.ToJson()))
 }
 
+// Deprecated: checkUserMfa is deprecated and should not be used anymore, starting with version 6.0 it will be disabled.
+//			   Clients should attempt a login without MFA and will receive a MFA error when it's required.
 func checkUserMfa(c *Context, w http.ResponseWriter, r *http.Request) {
+
+	if *c.App.Config().ServiceSettings.DisableLegacyMFA {
+		http.NotFound(w, r)
```

Comment:
I know this is already through review, but should this be a 501 Not Implemented rather than a 404? That's usually what we use when an endpoint is disabled due to config settings

Reply:
It would give a quite good indicator which 5.X version would be used, since in 6.0+ it will be a 404 since the route doesn't exist. Since it's deprecated and will not be turned back on once it's disabled in the long run, I would currently tend to stick to a Not Found to stay consistent with the behavior once we remove it completely.

Reply:
We expose the version number in the client config, so I don't think that part matters, but that makes sense to have it be a 404 since this is starting to remove that anyway.